src/memory.jl

```diff
@@ -38,9 +38,9 @@ function maybe_synchronize(managed::Managed)
     return nothing
 end
 
-function Base.convert(::Type{CLPtr{T}}, managed::Managed{M}) where {T, M}
+function Base.convert(t::Union{Type{CLPtr{T}}, Type{cl.cl_mem}}, managed::Managed{M}) where {T, M}
```

Comment:
Why do we need to extend this (and other) methods to support `cl_mem`? Don't we always use the device pointer that's part of the BDA, only using the underlying `cl_mem` to free etc?

Reply:
Oh right, the `memcpy` etc. methods are back-end specific (in CUDA we have simple `memcpy`s that support all kind of memory).